lib/frontdesk.rb

```diff
@@ -0,0 +1,85 @@
+require 'date'
+
+require_relative 'date.rb'
+require_relative 'reservation.rb'
+
+module Hotel
+  class FrontDesk
+    
+    attr_reader :all_reservations, :rooms
+    
+    def initialize
+      @all_reservations = all_reservations || []
+      @rooms = [*1..20]
+    end
+
+    def assign_room(new_reservation)
+      taken_rooms = []
+      all_rooms = @rooms.dup
+      @all_reservations.each do |reservation|
+        reservation.no_conflict?(new_reservation.start_date, new_reservation.end_date) ? taken_rooms << reservation.assigned_room : next
+      end
+      room_to_assign = (all_rooms - taken_rooms.flatten)
+      assigning = room_to_assign.sample(new_reservation.num_rooms)
+      taken_rooms.include?(assigning)
+        assigning = room_to_assign.sample(new_reservation.num_rooms)
+    end
+
+    def add_reservation(start_date, end_date, num_rooms)
+      new_reservation = Hotel::Reservation.new(start_date: start_date, end_date: end_date, num_rooms: num_rooms)
+      new_reservation.assigned_room = assign_room(new_reservation)
+      if new_reservation.num_rooms > 1 
+        raise ArgumentError.new("Invalid room reservation request: if you would still like #{num_rooms} rooms please reserve as a block.")
```

Comment:
:tada: Bad arguments in the `ArgumentError` message!  :tada:

I probably would have just left `num_rooms` off as an argument though, since it has to be `1`.